types/result.go

```diff
@@ -45,7 +45,7 @@ type ABCIMessageLogs []ABCIMessageLog
 
 // ABCIMessageLog defines a structure containing an indexed tx ABCI message log.
 type ABCIMessageLog struct {
-	MsgIndex int    `json:"msg_index"`
+	MsgIndex uint16 `json:"msg_index"`
```

Comment:
We change this to a `uint16` for two reasons:

1. We'll never exceed this anyway
2. We're guaranteed this is encoded and decoded as an integer and not stringified first (due to Amino)